feat: dataset downloader script

scottcanoe opened the pull request:
This PR introduces `download.py`, a script for downloading datasets for `tbp.tbs_sensorimotor_intelligence`.
As of now, it can download pretrained models for Monty, pre-computed results of experiments used in the paper, and the YCB dataset.

We'll have to modify this script when we make more datasets available, but that's pretty trivial.

In the below example
 - We already have the YCB dataset and DO NOT want to replace it.
 - We already have the pretrained models and DO want to replace them.
 - We don't have have the pre-computed results.
 
```shell
$ python download.py ycb monty.pretrained_models monty.results
ycb dataset already exists at '/Users/sknudstrup/tbp/data/habitat'. Overwrite? (y/[n]): n
monty.pretrained_models dataset already exists at '/Users/sknudstrup/tbp/results/dmc/pretrained_models'. Overwrite? (y/[n]): y
Downloading monty.pretrained_models dataset...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 3485M  100 3485M    0     0  34.8M      0  0:01:40  0:01:40 --:--:-- 36.0M
Downloading monty.results dataset...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 3230k  100 3230k    0     0  3651k      0 --:--:-- --:--:-- --:--:-- 3654k

```

nielsleadholm commented on download.py:
```diff
@@ -0,0 +1,216 @@
+  monty.view_finder_images
+                          Intermediate output used to generate input data
+                          for the ViT model (400 MB). Not required for
+                          running experiments or plotting scripts.
```
```suggestion
                          running Monty experiments or plotting scripts.
```

nielsleadholm reviewed (commented):
Really nice, thanks for putting this together, and @hlee9212 for the thorough review - agree with the points you raised but otherwise looks good to go.

scottcanoe commented:
Hey @nielsleadholm and @hlee9212, thanks for the review! I've implemented your suggestions. More specifically
 - We now have a -f/--force argument.
 - I've updated the datasets on AWS to have the .tgz extension (only) and their associated links in this script.
 - We now have exception handling in subprocess calls.
 
FYI, I've made both your changes manually (rather than the "Commit" button in GitHub).

nielsleadholm commented:
@scottcanoe is it possible to double check the AWS links? I'm getting an error when trying to download `monty.results`.
